features/filesystem/storagelite/source/StorageLiteFS.cpp

```
@@ -0,0 +1,320 @@
+    int ret = _stlite->init(_bd);
+    if (ret) {
+        return convert_to_errno(ret);
+    }
+
+    ret = _stlite->reset();
+    if (ret) {
+        return convert_to_errno(ret);
+    }
+
+    return 0;
+}
+
+int StorageLiteFS::remove(const char *filename)
+{
+    int ret = _stlite->remove((const uint8_t *) filename, strlen(filename) + 1);
+    return convert_to_errno(ret);
+}
+
+int StorageLiteFS::rename(const char *oldname, const char *newname)
+{
+    return -ENOSYS;
+}
+
+int StorageLiteFS::mkdir(const char *name, mode_t mode)
+{
+    return -ENOSYS;
+}
+
+int StorageLiteFS::stat(const char *name, struct stat *st)
+{
+    size_t file_size;
+    int ret = _stlite->get_file_size((const uint8_t *) name, strlen(name) + 1, file_size);
+    st->st_size = file_size;
+    return convert_to_errno(ret);
+}
+
+int StorageLiteFS::statvfs(const char *name, struct statvfs *st)
+{
+    return -ENOSYS;
+}
+
+int StorageLiteFS::file_open(mbed::fs_file_t *file, const char *path, int flags)
+{
+    file_handle_data_t *handle = new file_handle_data_t;
+    if (!handle) {
+        return -ENOMEM;
+    }
+
+    // We don't support opening for read-write, nor for appending
+    if ((flags & O_RDWR) || (flags & O_APPEND)) {
+        return -EINVAL;
+    }
+
+    if ((flags & 1) == O_RDONLY) {
+        // When open for reading, we need to check if file exists right now
+        int ret = _stlite->file_exists((const uint8_t *) path, strlen(path) + 1);
+        if (ret != STORAGELITE_SUCCESS) {
+            return convert_to_errno(ret);
+        }
+    } else {
+        if (flags & O_EXCL) {
+            int ret = _stlite->file_exists((const uint8_t *) path, strlen(path) + 1);
+            // Fail if file exists
+            if (ret == STORAGELITE_SUCCESS) {
+                return EEXIST;
+            }
+            if (ret != STORAGELITE_NOT_FOUND) {
+                return convert_to_errno(ret);
+            }
+        }
+    }
+
+    handle->file_name = new char[strlen(path) + 1];
```

needs nothrow if you want to return an error.